### PR TITLE
Don't hard code number of compile threads

### PR DESCRIPTION
**Pull request by bartoszcisek**
RVM can automatically detect number of available threads.

Changes proposed in this pull request:
* remove hardcoded `-j3`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #### Bug fixes:
 * $PATH become empty after __rvm_unload executed [\#3847](https://github.com/rvm/rvm/pull/3847)
 * RVM incorrectly tries to install llvm 3.5 when trying to install Rubinius 3  [\#3848](https://github.com/rvm/rvm/pull/3848), 
+* RVM hardcodes number of compile threads [\#3856](https://github.com/rvm/rvm/pull/3856)
```
Comment (mpapis) [inline CHANGELOG.md]: it's more: enable automated detection of compilation cores for compiling movable(binary) rubies

File: scripts/functions/build_config [hunk 2]
```diff
@@ -129,7 +129,6 @@ __rvm_setup_compile_environment_movable()
       ;;
   esac
   rvm_configure_flags+=( --sysconfdir=/etc )
-  rvm_make_flags+=( -j3 )
```
Comment (mpapis) [inline scripts/functions/build_config]: this enables automated detection of threads to the number of available cores

Comment (pkuczynski) [inline scripts/functions/build_config]: I agree with @bartoszcisek that this is obsolete and should be removed

File: CHANGELOG.md (hunk 1 above)
Comment (pkuczynski) [inline CHANGELOG.md]: This could be rephrased indeed, but its a minor thing

**Pull request merged**